src/Tribe/Editor/Hooks.php

```diff
@@ -27,6 +30,40 @@ public function register() {
 	 */
 	protected function add_actions() {
 		add_action( 'current_screen', [ $this, 'add_widget_resources' ] );
+		add_action( 'save_post_' . TEC::POSTTYPE, [ $this, 'calculate_duration' ], 20 );
+	}
+
+	/**
+	 * Calculate the event duration based on the saved start and end dates.
+	 * This occurs during post save, and checks the saved value against the calculated value on updates.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $post_id The event we are setting a duration for.
+	 */
+	public function calculate_duration( $post_id ) {
```

Comment:
The `Hooks` class is a service provider, it should not define or run business logic if possible. This method's business logic should be moved to a dedicated class.